### ISSUE TITLE
upgrade the embedded BatchingExecutor

### ISSUE THREAD

**Issue by patriknw** [akka/akka#26383]
Upgrading the embedded BatchingExecutor in the akka codebase to the new one here once it is merged: https://github.com/scala/scala/pull/7663/files#diff-4a66ae20fe1614a8edbd797892372d9f

thanks @viktorklang

**Comment by viktorklang** [comment]
The Scala update has been merged into the 2.13.x mainline, so this should be ready to import.

**Comment by jrudolph** [comment]
I don't think that's `small` though, as we'll probably have to do some benchmarking?

**Comment by patriknw** [comment]
fair point, changed to medium

**Comment by jrudolph** [comment]
Also see #26690, might make sense to answer the questions from that ticket in the progress.
